gossip/src/main.rs

```diff
@@ -283,14 +288,28 @@ fn process_rpc_url(
     let entrypoint_addr = parse_entrypoint(matches);
     let timeout = value_t_or_exit!(matches, "timeout", u64);
     let shred_version = value_t_or_exit!(matches, "shred_version", u16);
+
+    let gossip_host = parse_gossip_host(matches, entrypoint_addr);
+
+    let gossip_addr = SocketAddr::new(
+        gossip_host,
+        value_t!(matches, "gossip_port", u16).unwrap_or_else(|_| {
+            solana_net_utils::find_available_port_in_range(
+                IpAddr::V4(Ipv4Addr::UNSPECIFIED),
+                (0, 1),
+            )
+            .expect("unable to find an available gossip port")
+        }),
+    );
```

Comment:
can this code be refactored with the other place?

Reply:
ya np